noxfile.py

```diff
@@ -0,0 +1,37 @@
+"""Noxfile for performing various code checks."""
+
+import nox  # type: ignore
+
+locations = "pytket", "tests", "noxfile.py"
+
+"""Nox sessions."""
+
+
+@nox.session(python=["3.11", "3.10"])
+def test(session: nox.sessions.Session) -> None:
+    """Run the test suite."""
+    args = session.posargs or ["tests"]
+    session.install(".")
```

Comment:
is this locally installed as editable?

Reply:
It runs `python -m pip install .` Which installs from the local package, but I don't think as editable. But I don't think it needs to be editable because it's run at each invocation.

Reply:
When run with the `-r` option, nox will reuse the python environment and will only need to reinstall the package if changes were made. In that sense, the install process is usually very quick

Reply:
I am still wondering if it would be possible to do something like `session.install(". -e")` here to speed that up, but nox will run that again anyway if there are changes?

Reply:
It is possible, but in this case there is essentially no difference. As you say, it will then run `pip install -e .` every time, which takes the same (very small) amount of time.

Reply:
But I can use the `-e` option. It won't speed up anything with nox. But you could then activate the corresponding virtual environment in a shell and run the tests from there. In that case, it would be better to have installed with `-e`

Reply:
Changed it :)

Reply:
Great!